gridt/src/app/core/settings.service.ts

```diff
@@ -0,0 +1,44 @@
+import { Injectable } from "@angular/core";
+import { Observable, pipe, throwError, of, iif } from "rxjs";
+import { flatMap, tap, map, catchError, pluck, mergeMap, take } from "rxjs/operators";
+
+import { Identity } from './models/identity.model';
+import { AuthService } from './auth.service';
+import { SecureStorageService } from './secure-storage.service';
+
+@Injectable({
+  providedIn: "root"
+})
+
+export class SettingsService {
+
+  constructor (
+    private auth: AuthService,
+    private secStore: SecureStorageService,
+  ) {}
+
+  /**
+   * Observable to obtain settings from secure storage
+   * If secure storage is empty, this returns an error
+   */
+  public localIdentity$: Observable<Identity> = this.auth.readyAuthentication$.pipe(
+    flatMap(() => this.secStore.get$("identity"))
+  );
+
+  /**
+   * Store settings in the secure storage.
+   * @param identity Settings to be stored.
+   */
+  public setLocalIdentity$(identity: Identity): Observable<boolean> {
+    return this.auth.isLoggedIn$.pipe(
+      take(1),
+      tap( (val) => {
+        if (val === true) {
+          console.debug(`Storing identity: ${JSON.stringify(identity)} into the local storage.`);
+          this.secStore.set$("identity", identity).subscribe();
```

Comment:
This should be a flat map

https://www.learnrxjs.io/learn-rxjs/operators/transformation/mergemap

Reply:
They are the same. The website says mergeMap is an alias for flatMap. So why is this wrong?

Reply:
Ooh, sorry I was looking at the wrong place for this comment.

Reply:
Why? I don't want the output of secStore.set$ to be the new observable of setLocalIdentity$.

Reply:
The output of isLoggedIn is a Boolean and after observing that you are, this value is no longer important. Once that step is completed, the only thing we are still interested in, is if the settings were stored properly. If they were we can return true. If not then we throw an error. Now the error catching will have to be done in a way that it is able to distinguish between an error caused by a failed login and a failed setting, but all of this is clearly very descriptive in the sense that rxjs is useful for describing how it should be done. 


Note: in the current state you don't do anything with the case that the user is not loggedIn, it would be good to throw an error when he is not loggedIn so that for the observer it is clear that it failed and why.